flow_runner.go

```diff
@@ -24,20 +24,55 @@ type flowRunner struct {
 
 // Run runs provided tasks and all their dependencies.
 // Each task is executed at most once.
-func (f *flowRunner) Run(ctx context.Context, args []string) int { //nolint // TODO: refactor
+func (f *flowRunner) Run(ctx context.Context, args []string) int {
+	f.verifyAllParametersAreInUse()
+	f.initializeParameters()
+	tasks, usageRequested, err := f.parseArguments(args)
+	if err != nil {
+		fmt.Fprintf(f.output, "cannot parse arguments: %v\n", err)
+		return CodeInvalidArgs
+	}
+
+	if usageRequested {
+		printUsage(f)
+		return CodePass
+	}
+
+	tasks = f.defaultTasks(tasks)
```

Comment:
this method name is not readable to me

maybe we could simply inline `defaultTasks(tasks)` implementation in the bottom of `parseArguments` method?

Reply:
`parseArguments` is not responsible for the business decision of defaulting tasks.

In my mind this step that is currently (only) resolving which task to run by default is possibly one that generally handles the list of requested tasks. For instance, at one point this "phase" could be extended to sort the order of execution of tasks (bogus example) - I was thinking in terms of the idea we had about plugins.
So, this is somewhat a phase like "process requested tasks" - though I didn't follow up on finding a proper name, as this would be YAGNI at this point. Hence I called it what it only does: task defaulting.

Reply:
how about `tasksToRun` ? 😄

Reply:
Good idea - code changed